### PR TITLE
fix: ensure default_plugins/0 is the same as loaded_plugins.tmpl.

### PR DESCRIPTION
**Pull request by zhongwencool**
Fixes [EMQX-9660](https://emqx.atlassian.net/browse/EMQX-9660)

No changelog is needed  there were no modifications made to the open-source version, only tests were added.
Add changelog to EE later.

### <samp>🤖 Generated by Copilot at 0ddc9ba</samp>

This pull request updates the `emqx` application and the `emqx_plugins` module to improve the plugin management functionality and consistency across versions. It also changes the default values of some plugins to match the latest configuration files. It modifies the `emqx.appup.src`, `emqx.app.src`, `emqx_plugins.erl`, and `emqx_plugins_SUITE.erl` files.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/{ce,ee}/(feat|perf|fix)-<PR-id>.en.md` files
- [ ] For internal contributor: there is a jira ticket to track this change
- [ ] If there should be document changes, a PR to emqx-docs.git is sent, or a jira ticket is created to follow up
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

**Comment by terry-xiaoyu** [comment]
Please update the appup files, and the change-log (add `changes/e4.4.18-en.md` and `changes/e4.4.18-zh.md` files)

**Comment by zhongwencool** [comment]
> Please update the appup files, and the change-log (add `changes/e4.4.18-en.md` and `changes/e4.4.18-zh.md` files)

Shouldn't a changelog be included in enterprise versions? or this  repo ?

File: test/emqx_plugins_SUITE.erl [hunk 1]
```diff
@@ -125,9 +128,23 @@ t_ensure_default_loaded_plugins_file(Config) ->
     ok = emqx_plugins:load(),
     {ok, Contents} = file:consult(TmpFilepath),
     DefaultPlugins = default_plugins(),
-    ?assertEqual(DefaultPlugins, lists:sort(Contents)),
+    ?assertEqual(lists:sort(DefaultPlugins), lists:sort(Contents)),
+
+    GenContents = get_loaded_plugins_from_tmpl(),
+    Fun = fun({Name, _}) -> Name end,
```
Comment (JimMoen) [inline test/emqx_plugins_SUITE.erl:134]: The plugin `enable/disabled` state should be checked too.

**Comment by terry-xiaoyu** [comment]
Please rebase this PR to release-v44, as the main-v4.4 branch is for changes target to 4.4.19.

File: src/emqx_plugins.erl [hunk 2]
```diff
@@ -252,11 +252,14 @@ default_plugins() ->
...
-        {emqx_recon, true},
+        {emqx_recon, false},
         %% emqx_telemetry is not exist in enterprise.
         %% {emqx_telemetry, false},
         {emqx_rule_engine, true},
-        {emqx_bridge_mqtt, false}
+        {emqx_bridge_mqtt, false},
+        {emqx_schema_registry, true},
+        {emqx_eviction_agent, true},
+        {emqx_node_rebalance, true}
```
[the remark this answers is not in the document]
Comment (zhongwencool) [reply src/emqx_plugins.erl:262]: Not really, recon is enabled in open source, but disabled on ee.

https://github.com/emqx/emqx-enterprise/blob/main-v4.4-enterprise/rebar.config.erl#L597-L606

File: test/emqx_plugins_SUITE.erl (hunk 1 above)
Comment (zhongwencool) [reply test/emqx_plugins_SUITE.erl]: DONE

**Review by terry-xiaoyu — APPROVED** [review]

**Pull request merged**